test/unittests.jl

```diff
@@ -2000,6 +2002,126 @@ end
+    @test convert(Quantity{Float64, AffineDimensions{Float64}}, u"kg") isa Quantity{Float64, AffineDimensions{Float64}}
+    @test convert(Quantity{Float64, Dimensions}, ua"kg") isa Quantity{Float64, Dimensions{DynamicQuantities.DEFAULT_DIM_BASE_TYPE}}
+    
+    # Test uncovered operations
+    @test (2.0ua"m")^2 == (2.0u"m")^2
+    @test dimension(ua"m")^Int32(2) == dimension(ua"m^2")
+    @test 2.0u"m" + 2.0ua"m" === 4.0u"m"
+    @test 2.0ua"m" + 2.0ua"m" === 4.0u"m"
+    @test 2.0u"m" - 2.0ua"m" === 0.0u"m"
+    @test 2.0ua"m" - 2.0ua"cm" === 1.98u"m"
+    @test 5.0°C - 4.0°C === 1.0u"K"
+    @test mod(2.0u"m", 2.0ua"m") === 0.0u"m"
+    @test mod(2.0ua"m", 2.0ua"m") === 0.0u"m"
+    @test_throws AffineOffsetError mod(2.0ua"°C", 2.0ua"°C")
+
+    @test 2.0u"K" ≈ 2.0ua"K"
+    @test 2.0ua"K" ≈ 2.0ua"K"
+    @test 2.0ua"K" ≈ 2.0u"K"
+    @test_throws AffineOffsetError (2ua"°C")^2
+    @test uexpand(2ua"°C") == 275.15u"K"
+
+    # Test conversions
+    @test °C |> us"K" isa Quantity{<:Real, <:SymbolicDimensions}
+    @test 0°C |> us"K" == 273.15us"K"
+    @test us"K" |> °C isa Quantity{<:Real, <:AffineDimensions}
+    @test 0us"K" |> °C  == -273.15°C
+    @test °C |> °F isa Quantity{<:Real, <:AffineDimensions}
+    @test 0°C |> °F == 32°F
+
+    @test QuantityArray([0,1]°C) |> °F isa QuantityArray{T, <:Any, AffineDimensions{R}} where {T,R}
+
+    # Test display against errors
+    celsius = AffineDimensions(offset=273.15, basedim=u"K")
+    psi = AffineDimensions(basedim=6.89476us"kPa")
+    io = IOBuffer()
+    @test isnothing(show(io, (dimension(°F), dimension(ua"K"), psi, celsius, fahrenheit)))
+
+    # Test updating affine units
+    @test isnothing(DynamicQuantities.update_external_affine_unit(:°C, °C)) #same value yields nothing for quantity
+    @test isnothing(DynamicQuantities.update_external_affine_unit(:°C, dimension(°C))) #same value yields nothing for dimension with symbol
+    @test isnothing(DynamicQuantities.update_external_affine_unit(dimension(°C))) #same value yields for dimension
+    @test_throws "Cannot register a unit if its symbol is :nothing" DynamicQuantities.update_external_affine_unit(celsius) #cannot register :nothing
+
+    #Cannot re-register a unit if its value changes
+    @test_throws "Unit `°C` already exists as `(scale = 1.0, offset = 273.15, basedim = K)`, its value cannot be changed to `(scale = 2.0, offset = 273.15, basedim = K)`" DynamicQuantities.update_external_affine_unit(:°C, 2*°C)
+
+    # Test map_dimensions
+    @test map_dimensions(+, dimension(ua"m/s"), dimension(ua"m/s")) == AffineDimensions(Dimensions(length=2, time=-2))
+    @test map_dimensions(-, dimension(ua"m"), dimension(ua"s"))     == AffineDimensions(Dimensions(length=1, time=-1))
+    @test map_dimensions(Base.Fix1(*,2), dimension(ua"m/s"))        == AffineDimensions(Dimensions(length=2, time=-2))
```

Comment:
Please avoid these kinds of stylistic patterns with multiple spaces to line up the `==`